GitUI/Theming/ThemeController.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Reflection;
+using System.Windows.Forms;
+using GitCommands;
+using GitExtUtils.GitUI.Theming;
+
+namespace GitUI.Theming
+{
+    public class ThemeController : Theme
+    {
+        private readonly Theme _defaultTheme;
+        private readonly FieldInfo _colorTableField;
+        private readonly PropertyInfo _threadDataProperty;
+        private readonly object _systemBrushesKey;
+        private readonly object _systemPensKey;
+        private readonly Dictionary<KnownColor, Color> _sysColors;
+
+        public ThemeController(Theme defaultTheme)
+        {
+            _defaultTheme = defaultTheme;
+            _sysColors = new Dictionary<KnownColor, Color>();
+
+            var systemDrawingAssembly = typeof(Color).Assembly;
+
+            _colorTableField = systemDrawingAssembly.GetType("System.Drawing.KnownColorTable")
+                .GetField("colorTable", BindingFlags.Static | BindingFlags.NonPublic);
+
+            _threadDataProperty = systemDrawingAssembly.GetType("System.Drawing.SafeNativeMethods")
+                .GetNestedType("Gdip", BindingFlags.NonPublic)
+                .GetProperty("ThreadData", BindingFlags.Static | BindingFlags.NonPublic);
+
+            _systemBrushesKey = typeof(SystemBrushes)
+                .GetField("SystemBrushesKey", BindingFlags.Static | BindingFlags.NonPublic)
+                .GetValue(null);
+
+            _systemPensKey = typeof(SystemPens)
+                .GetField("SystemPensKey", BindingFlags.Static | BindingFlags.NonPublic)
+                .GetValue(null);
+        }
+
+        public event Action Changed;
+
+        private IDictionary ThreadData =>
+            (IDictionary)_threadDataProperty.GetValue(null, null);
+
+        public void Load(
```

Comment:
I am confused, what is the purpose of this method?

Reply:
renamed to `SetColors`

Reply:
Could you please add an xml-doc explaining what the API does, it would help with comprehension.
Thank you

Reply:
documented `ThemeManager` and it's base class `Theme`